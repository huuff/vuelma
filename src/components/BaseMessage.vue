<template>
<article class="message" v-show="show" :class="`is-${color}`">
  <header class="message-header">
    <p v-if="title">{{ title }}</p>
    <slot name="header"></slot>
    <button v-if="closeable"
      class="delete"
      aria-label="delete"
      @click="$emit('close')"
    ></button>
  </header>
  <div class="message-body">
    <slot></slot>
  </div>
</article>
</template>

<script setup lang="ts">
import { useSlots, } from 'vue';
import { BulmaColor } from '@/types/bulma-color';

const props = withDefaults(defineProps<{
  title?: string;
  closeable?: boolean;
  show?: boolean;
  color?: BulmaColor;
}>(), {
  closeable: false,
  show: true,
  color: 'dark',
});

const emit = defineEmits<{
  (event: 'close'): void
}>();

const slots = useSlots();

if (props.title && slots.header) {
  throw new Error("In a BaseMessage component, can't have a title and a header at the same time!")
}
</script>
