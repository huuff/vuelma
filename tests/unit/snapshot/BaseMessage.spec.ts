import { mount } from "@vue/test-utils"
import BaseMessage from "@/components/BaseMessage.vue";

test("BaseMessage.vue snapshot", () => {
  const wrapper = mount(BaseMessage, {
    props: {
      titleText: "Message title",
      closeable: true,
      show: true,
      color: "info"
    },
    slots: {
      default: `
        <p>Message contents!</p>
      `
    },
  });

  expect(wrapper.element).toMatchSnapshot();
})
