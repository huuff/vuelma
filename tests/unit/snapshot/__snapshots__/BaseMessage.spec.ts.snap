// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseMessage.vue snapshot 1`] = `
<article
  class="message is-info"
>
  <header
    class="message-header"
  >
    <p>
      Message title
    </p>
    
    
    <button
      aria-label="delete"
      class="delete"
    />
  </header>
  <div
    class="message-body"
  >
    
    <p>
      Message contents!
    </p>
    
  </div>
</article>
`;
